test/SparqlAntlrParserTest.cpp

```diff
@@ -1165,8 +1184,50 @@ TEST(SparqlParser, GroupGraphPattern) {
                                             INTERNAL_PREDICATE_PREFIX_IRI}}),
         matcher);
   };
+
+  // Test the Components alone.
+  expectGraphPattern("{ }", GraphPattern(false, {}));
+  expectGraphPattern(
+      "{ { ?a ?b ?c } }",
+      GraphPattern(
+          false, {},
+          IsGroup(GraphPattern(false, {}, Triples({{"?a", "?b", "?c"}})))));
+  expectGraphPattern(
+      "{ { ?a ?b ?c } UNION { ?d ?e ?f } }",
+      GraphPattern(
+          false, {},
+          IsUnion(GraphPattern(false, {}, Triples({{"?a", "?b", "?c"}})),
+                  GraphPattern(false, {}, Triples({{"?d", "?e", "?f"}})))));
+  expectGraphPattern(
+      "{ { ?a ?b ?c } UNION { ?d ?e ?f } UNION { ?g ?h ?i } }",
+      GraphPattern(
+          false, {},
+          IsUnion(
+              GraphPattern(
+                  false, {},
+                  IsUnion(
+                      GraphPattern(false, {}, Triples({{"?a", "?b", "?c"}})),
+                      GraphPattern(false, {}, Triples({{"?d", "?e", "?f"}})))),
+              GraphPattern(false, {}, Triples({{"?g", "?h", "?i"}})))));
+  expectGraphPattern(
+      "{ OPTIONAL { ?a <foo> <bar> } }",
+      GraphPattern(false, {},
+                   Optional({}, Triples({{"?a", "<foo>", "<bar>"}}))));
+  expectGraphPattern(
+      "{ MINUS { ?a <foo> <bar> } }",
+      GraphPattern(false, {}, Minus({}, Triples({{"?a", "<foo>", "<bar>"}}))));
+  expectGraphPattern(
+      "{ FILTER (?a = 10) }",
+      GraphPattern(false, {{SparqlFilter::FilterType::EQ, "?a", "10"}}));
+  expectGraphPattern("{ BIND (?f - ?b as ?c) }",
+                     GraphPattern(false, {}, Bind("?c", "?f-?b")));
+  expectGraphPattern("{ VALUES (?a ?b) { (<foo> <bar>) (<a> <b>) } }",
```

Comment:
I would leave this for now, and will merge these tests as is, 
but for the next round we should consider a variant of those matches without having to specify `false, {}` which are almost always there and increase the visual noise.